theme/lumo/vaadin-confirm-dialog-styles.html

```diff
@@ -56,15 +56,18 @@
 
       .cancel-button {
         flex-grow: 1;
+      }
+
+      :host(:not([dir="rtl"])) .cancel-button {
         margin-left: calc(var(--lumo-space-s) * -1);
       }
 
-      .confirm-button {
+      :host(:not([dir="rtl"])) .confirm-button {
```

Comment:
Hmm.. that's tricky. The way `vaadin-confirm-dialog` uses `vaadin-dialog` creates a shadow root on the dialog's `[part=content]` div, and the styles targeting `:host` are not working, so all the styles with that are not applied.

Reply:
With that, `vaadin-overlay` changes all `:host` selectors to `:host-nomatch`, so it will not work. I couldn't find a proper way to make the elements of the confirm dialog aware of the `dir`.

The only one that came to me would be by setting `dir` to the elements we want to target either with `attributeChangedCallback` (for the case when a change to `dir` happens when dialog is opened) and at the `_openedChanged` method (for the case when the dialog gets opened).

Reply:
I was searching for the solution yesterday and today and came up with something similar